src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -1563,6 +1577,118 @@ def _get_linux_multicontainer_encoded_config_from_file(file_name):
+        function_app_config["deployment"]["storage"] = functionapp_deployment_storage = {"type": "blobContainer", "value": None, "authentication": None}
+        
+    if ("authentication" not in functionapp_deployment_storage or functionapp_deployment_storage["authentication"] is None):
+        if (deployment_storage_auth_type == None):
+            raise ValidationError("Please provide a value for --deployment-storage-auth-type as function app deployment storage authentication type is not set.")
+        
+        functionapp_deployment_storage["authentication"] = {"type": "SystemAssignedIdentity", "userAssignedIdentityResourceId": None, "storageAccountConnectionStringName": None}
+
+    # Storage
+    deployment_config_storage_value = None
+    if (deployment_storage_name is not None):
+        deployment_storage = _validate_and_get_deployment_storage(cmd.cli_ctx, resource_group_name, deployment_storage_name)
+        deployment_storage_container = _get_deployment_storage_container(cmd, resource_group_name, deployment_storage_name, deployment_storage_container_name)
+        
+        if deployment_storage_container is None:
+            raise ValidationError("The specified container does not exist in the specified storage account.")
+        
+        deployment_storage_container_name = deployment_storage_container.name
+        deployment_config_storage_value = getattr(deployment_storage.primary_endpoints, 'blob') + deployment_storage_container_name
+        functionapp_deployment_storage["value"] = deployment_config_storage_value
+        functionapp_deployment_storage["type"] = "blobContainer"
+        
+    # Authentication
+    if (deployment_storage_auth_type != None):
+        assign_identities = None
+        if deployment_storage_auth_type == 'storageAccountConnectionString':
+            deployment_storage_conn_string = _get_storage_connection_string(cmd.cli_ctx, deployment_storage)
+            configs = get_site_configs(cmd, resource_group_name, name, slot)
+            
+            if configs.app_settings is None:
+                configs.app_settings = []
+            configs.app_settings.append(NameValuePair(name='DEPLOYMENT_STORAGE_CONNECTION_STRING',
+                                                            value=deployment_storage_conn_string))
+            update_flex_functionapp_configuration(cmd, resource_group_name, name, configs)
+            
+            functionapp_deployment_storage["authentication"]["type"] = "StorageAccountConnectionString"
+            functionapp_deployment_storage["authentication"]["userAssignedIdentityResourceId"] = None
+            functionapp_deployment_storage["authentication"]["storageAccountConnectionStringName"] = "DEPLOYMENT_STORAGE_CONNECTION_STRING"
+        elif deployment_storage_auth_type == 'systemAssignedIdentity':
+            assign_identities = ['[system]']
+            functionapp_deployment_storage["authentication"]["type"] = "SystemAssignedIdentity"
```

Comment:
It has to be either
```suggestion
            functionapp_deployment_storage["authentication"]["type"] = "systemAssignedIdentity"
```
or 
```suggestion
            functionapp_deployment_storage["authentication"]["type"] = deployment_storage_auth_type
```

Reply:
I think backend is expecting `SystemAssignedIdentity` instead of `systemAssignedIdentity`